src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Jlink.java

```diff
@@ -164,12 +165,13 @@ public JlinkConfiguration(Path output,
                                   ModuleFinder finder) {
             this.output = output;
             this.modules = Objects.requireNonNull(modules);
-            this.endian = Objects.requireNonNull(endian);
+            this.endian = endian;
```

Comment:
`JlinkConfiguration` does not need to know the endianness.   Only the image builder needs it.   Probably good to take the `endian` parameter out from `JlinkConfiguration` and instead pass the value of `--endian` to `JlinkTask::createImageProvider` like other jlink options.

Reply:
Done. The updated PR no longer has the endian value in the `JlinkConfiguration`